### DISCUSSION TITLE
Resource exhauster error when trying to insert embeddings in Milvus collection

### DISCUSSION THREAD

**Discussion by ZubeenK** [milvus-io/milvus#23888] [Q&A and General discussion] [answered]
I am getting RPC error: ERROR: RPC error: [batch_insert], <MilvusException: (code=1, message=<_MultiThreadedRendezvous of RPC that terminated with:
	status = StatusCode.RESOURCE_EXHAUSTED
	details = "grpc: received message larger than max (132320047 vs. 67108864)"

I am using the below code to insert embeddings into vector collection. The size of embeddings is around 284MB.
def insert_data(collection_name, vectors_list):
    res = collection_name.insert(vectors_list)
    print(f"{res} = data insertion completed sucessfully")
    
pymilvus==2.2.1 version
How can I resolve this problem?

**Comment by xiaofan-luan** [comment, marked as answer]
> I am getting RPC error: ERROR: RPC error: [batch_insert], <MilvusException: (code=1, message=<_MultiThreadedRendezvous of RPC that terminated with: status = StatusCode.RESOURCE_EXHAUSTED details = "grpc: received message larger than max (132320047 vs. 67108864)"
> 
> I am using the below code to insert embeddings into vector collection. The size of embeddings is around 284MB. def insert_data(collection_name, vectors_list): res = collection_name.insert(vectors_list) print(f"{res} = data insertion completed sucessfully")
> 
> pymilvus==2.2.1 version How can I resolve this problem?

insert batch is too large. each batch should be less than 64M. try to split embeddings and insert.

Also, recommended to use 2.2.8, this will be more stable

**Reply by ghost** [reply]
#- [ ]
